app/src/main/java/com/github/onlynotesswent/ui/overview/FolderContent.kt

```diff
@@ -96,7 +95,9 @@ fun FolderContentScreen(
 
   // Custom back handler to manage back navigation
   BackHandler {
-    if (folder.value!!.parentFolderId != null) {
+    if (!folder.value!!.isOwner(currentUser.value!!.uid)) {
+      navigationActions.navigateTo(TopLevelDestinations.SEARCH)
```

Comment:
For the moment public folders only give access to notes, subfolders don't display to prevent users from accessing all subfolder hierarchy of another user